src/main/httpCommandServer.ts

```diff
@@ -0,0 +1,103 @@
+import fs from 'fs';
+import http from 'http';
+import path from 'path';
+
+import Logging from '@/utils/logging';
+import paths from '@/utils/paths';
+
+type ServerState = {
+  user: string;
+  password: string;
+  port: number;
+  pid: number;
+}
+
+const console = Logging.server;
+const ServerPort = 6107;
+const ServerUsername = 'user';
+const ServerFileBasename = 'rd-engine.json';
+
+export default class HttpCommandServer {
+  protected server = http.createServer();
+  protected password = randomStr();
+  protected stateInfo: ServerState = {
+    user:     ServerUsername,
+    password: this.password,
+    port:     ServerPort,
+    pid:      process.pid,
+  };
+
+  async init() {
+    const statePath = path.join(paths.appHome, ServerFileBasename);
+
+    await fs.promises.writeFile(statePath,
+      JSON.stringify(this.stateInfo, undefined, 2),
+      { mode: 0o600 });
+    this.server.listen(ServerPort, '127.0.0.1');
+    console.log(`Listening on port ${ ServerPort }, user: ${ ServerUsername },  password: ${ this.password }`);
+    this.server.on('request', this.handleRequest.bind(this));
+    this.server.on('error', (err) => {
+      console.log(`Error: ${ err }`);
```

Comment:
Should we include the stack? (That is, you're converting `err` to a string here, which doesn't include the stack.)

Reply:
The stack isn't very useful these days because all the code gets converted. When we get to building a jsonapi interface we'll most likely scope out the error fields we want to return. For now logging the stringified error message should be sufficient.